core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -826,12 +852,31 @@ public RexNode makeAbstractCast(RelDataType type, RexNode exp) {
    * @return Call to CAST operator
    */
   public RexNode makeAbstractCast(RelDataType type, RexNode exp, boolean safe) {
-    SqlOperator operator =
+    final SqlOperator operator =
         safe ? SqlLibraryOperators.SAFE_CAST
             : SqlStdOperatorTable.CAST;
     return new RexCall(type, operator, ImmutableList.of(exp));
   }
 
+  /**
+   * Creates a call to CAST or SAFE_CAST operator with a FORMAT clause.
+   *
+   * @param type Type to cast to
+   * @param exp  Expression being cast
+   * @param safe Whether to return NULL if cast fails
+   * @param format Conversion format for target type
+   * @return Call to CAST operator
+   */
+  public RexNode makeAbstractCast(RelDataType type, RexNode exp, boolean safe, RexLiteral format) {
```

Comment:
does this need to be nullable

Reply:
If you're referring to the format argument being null, we're creating a null valued RexLiteral object if we receive an empty format string from the parser, so it doesnt need to be nullable.

Reply:
oh duh